src/zope/securitypolicy/grantinfo.py

```diff
@@ -36,42 +36,35 @@
 @implementer(IGrantInfo)
 class AnnotationGrantInfo(object):
 
-    prinper = prinrole = permrole = {}
-
     def __init__(self, context):
         self._context = context
-        annotations = IAnnotations(context, None)
-        if annotations is not None:
+        annotations = IAnnotations(context, {})
 
-            prinper = annotations.get(prinperkey)
-            if prinper is not None:
-                self.prinper = prinper._bycol  # by principals
+        # by principals
+        prinper = annotations.get(prinperkey)
+        self.prinper = prinper._bycol if prinper is not None else {}
 
-            prinrole = annotations.get(prinrolekey)
-            if prinrole is not None:
-                self.prinrole = prinrole._bycol  # by principals
+        # by principals
+        prinrole = annotations.get(prinrolekey)
+        self.prinrole = prinrole._bycol if prinrole is not None else {}
 
-            roleper = annotations.get(rolepermkey)
-            if roleper is not None:
-                self.permrole = roleper._byrow  # by permission
+        # by permission
+        roleper = annotations.get(rolepermkey)
+        self.permrole = roleper._byrow if roleper is not None else {}
 
-    def __nonzero__(self):
+    def __bool__(self):
         return bool(self.prinper or self.prinrole or self.permrole)
 
+    __nonzero__ = __bool__
+
     def principalPermissionGrant(self, principal, permission):
-        prinper = self.prinper.get(principal)
-        if prinper:
-            return prinper.get(permission, Unset)
-        return Unset
+        prinper = self.prinper.get(principal, {})
+        return prinper.get(permission, Unset)
 
     def getRolesForPermission(self, permission):
-        permrole = self.permrole.get(permission)
-        if permrole:
-            return permrole.items()
-        return ()
+        permrole = self.permrole.get(permission, {})
+        return list(permrole.items())
 
     def getRolesForPrincipal(self, principal):
-        prinrole = self.prinrole.get(principal)
-        if prinrole:
-            return prinrole.items()
-        return ()
+        prinrole = self.prinrole.get(principal, {})
+        return list(prinrole.items())
```

Comment:
I'm not sure that's really a problem because (1) The only references to either `IGrantInfo` (the only interface this class implements, which does not extend anything except `Interface`) or `AnnotationGrantInfo` across both zopefoundation and plone organizations are in this repository; and (2) it was completely untested before this. So I don't think it's really used.

Plus, Python 2 is going away so the number of affected users is small and getting smaller 😄 

But if you feel it's important I will happily make a change.

Reply:
Nah, your arguments sound very convincing.